src/foam/nanos/http/SessionWebAgent.java

```diff
@@ -83,13 +83,14 @@ public void execute(X x) {
       // check permissions
       Subject subject = new Subject.Builder(x).setUser(user).build();
       session.setContext(session.getContext().put("subject", subject));
-      if ( ! auth.check(session.getContext(), permission_) ) {
+      var applyX = session.applyTo(x);
+      if ( ! auth.check(applyX, permission_) ) {
         throw new AuthorizationException();
       }
 
       // execute delegate
       // Update session context with support setup from earlier WebAgents.
-      getDelegate().execute(session.getContext().put(HttpServletResponse.class, resp).put(HttpServletRequest.class, req));
+      getDelegate().execute(applyX.put(HttpServletResponse.class, resp).put(HttpServletRequest.class, req));
```

Comment:
This is not necessary after applyTo as applyTo will return null if they are not already in the context.